src/client/Microsoft.Identity.Client/Platforms/Android/Broker/AndroidContentProviderBroker.cs

```diff
@@ -0,0 +1,349 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Android.App;
+using Android.Content;
+using Android.Database;
+using Android.OS;
+using Microsoft.Identity.Client.ApiConfig.Parameters;
+using Microsoft.Identity.Client.Core;
+using Microsoft.Identity.Client.Internal.Broker;
+using Microsoft.Identity.Client.Internal.Requests;
+using Microsoft.Identity.Client.OAuth2;
+using Microsoft.Identity.Client.UI;
+using AndroidNative = Android;
+using AndroidUri = Android.Net.Uri;
+
+namespace Microsoft.Identity.Client.Platforms.Android.Broker
+{
+    [AndroidNative.Runtime.Preserve(AllMembers = true)]
+    internal class AndroidContentProviderBroker : IBroker
+    {
+        private readonly AndroidBrokerHelper _brokerHelper;
+        private readonly ICoreLogger _logger;
+        private Activity _parentActivity;
+        private string _negotiatedBrokerProtocalKey = String.Empty;
+
+        public AndroidContentProviderBroker(CoreUIParent uiParent, ICoreLogger logger)
+        {
+            _parentActivity = uiParent?.Activity;
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            AuthenticationContinuationHelper.LastRequestLogger = _logger;
+            _brokerHelper = new AndroidBrokerHelper(Application.Context, logger);
+        }
+
+        public bool IsBrokerInstalledAndInvokable()
+        {
+            return _brokerHelper.IsBrokerInstalledAndInvokable();
+        }
+
+        public async void InitiateBrokerHandshakeAsync()
+        {
+            using (_logger.LogMethodDuration())
+            {
+                try
+                {
+                    Bundle HandshakeBundleResult = await GetHandshakeBundleResultFromBrokerAsync().ConfigureAwait(false);
+
+                    _negotiatedBrokerProtocalKey = _brokerHelper.GetProtocalKeyFromHandshakeResult(HandshakeBundleResult);
```

Comment:
Please install an extension like 
![image](https://user-images.githubusercontent.com/12273384/106800400-4ecac980-6658-11eb-8865-672b7c13d1ea.png)

It will help auto-correct these quickly typed names. I have this problem as well.

`Protocol`

Reply:
Definitely need this